packages/protocol/src/peer/PeerStore.ts

```diff
@@ -16,4 +18,25 @@ export abstract class PeerStore {
     abstract loadPeers(): MaybePromise<Iterable<OperationalPeer>>;
     abstract updatePeer(peer: OperationalPeer): MaybePromise<void>;
     abstract deletePeer(address: PeerAddress): MaybePromise<void>;
+    abstract createNodeStore(address: PeerAddress): Promise<PeerNodeStore>;
+}
+
+export abstract class PeerNodeStore {
+    abstract construction: Construction<PeerNodeStore>;
+
+    abstract maxEventNumber: EventNumber;
+    abstract updateLastEventNumber(eventNumber: EventNumber): MaybePromise<void>;
+
+    abstract retrieveAttribute(
+        endpointId: EndpointNumber,
+        clusterId: ClusterId,
+        attributeId: AttributeId,
+    ): DecodedAttributeReportValue<any> | undefined;
+    abstract retrieveAttributes(endpointId: EndpointNumber, clusterId: ClusterId): DecodedAttributeReportValue<any>[];
```

Comment:
I though and tried. The main topic is that we likely would need a proxy to handle direct access to such a structure without generating the whole data. Using a "getter" to dynamically build it would be too much if we just need to access a single value (aka endpoint - cluster-attribute path).
And if we need a function for the direct access then a function for "get all" which can do additional convenient filtering as needed" is not much more. I added a todo. In my eyes we can revisit this with Client Node and once we remove legacy API or such. If you have an idea how to best do it before without having a too huge performance impact tell me :-)